test/common/http/http2/http2_frame_test.cc

```diff
@@ -0,0 +1,34 @@
+#include <string>
+
+#include "envoy/http/metadata_interface.h"
+
+#include "common/http/http2/metadata_encoder.h"
+
+#include "test/common/http/http2/http2_frame.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http2 {
+// From metadata map
+TEST(EqualityMetadataFrame, Http2FrameTest) {
+  MetadataMap metadataMap = {{"Connections", "15"}, {"Timeout Seconds", "10"}};
+  Http2Frame http2FrameFromUtility = Http2Frame::makeMetadataFrameFromMetadataMap(
+      1, metadataMap, Http2Frame::MetadataFlags::EndMetadata);
+  MetadataEncoder metadataEncoder{};
+  Http::MetadataMapPtr metadataMapPtr = std::make_unique<Http::MetadataMap>(metadataMap);
+  Http::MetadataMapVector metadata_map_vector;
+  metadata_map_vector.push_back(std::move(metadataMapPtr));
+  metadataEncoder.createPayload(metadata_map_vector);
+  std::string payloadFromEncoder = metadataEncoder.payload();
```

Comment:
since we dont direct compare bytes any more with this payload, maybe this, the encoder, and the `payload()` method can be removed

Reply:
+ `payload()`